src/qibo/models/evolution.py

```diff
@@ -129,8 +143,15 @@ def execute(self, final_time, start_time=0.0, initial_state=None):
             if self.callbacks:
                 state = self.normalize_state(state)
                 self.calculate_callbacks(state)
+
+            if self.track_state:
+                state_hist.append(state)
```

Comment:
This will not work properly if a qibojit operation is used, which can happen if you use `SymbolicHamiltonian` and Trotterization, which falls back to circuit execution. Qibojit updates the state in-place so in the end you'll get a list of references to the same array, which will be the final state only.